libvast/vast/optional.hpp

```diff
@@ -15,10 +15,20 @@
 
 #include <caf/optional.hpp>
 
+#include <optional>
+
 namespace vast {
 
 /// A drop-in replacement for C++17's std::optional.
 using caf::optional;
 
+template <typename T>
+std::optional<T> to_std(caf::optional<T>&& opt) {
```

Comment:
```suggestion
std::optional<T> to_std(caf::optional<T> opt) {
```

Anything against this?

Reply:
It would work as well, but it leaves the possibility to accidentally make huge copies without noticing due to user error.

Reply:
I'm fine with starting out with rvalue reference. In general, having values as parameters gives flexibility on the call site. For example, I would want to pass an `optional<int>` by value.

Reply:
If you want to keep using the original optional after the call you can still do that, the copy just needs to be made explicitly:

```
caf::optional<int> copt;
auto sopt = to_std(caf::optional<int>{copt});
```

While its technically not possible to actually pass by value using only this overload, for types like `int` that don't define a separate move constructor, this is ultimately going to reduce to using the copy constructor anyways so the generated code will be the same.